src/ucs/arch/x86_64/cpu.c

```diff
@@ -462,10 +462,13 @@ static size_t ucs_cpu_memcpy_thresh(size_t user_val, size_t auto_val)
 void ucs_cpu_init()
 {
 #if ENABLE_BUILTIN_MEMCPY
+    size_t l3_cache = ucs_cpu_get_cache_size(UCS_CPU_CACHE_L3);
+
     ucs_global_opts.arch.builtin_memcpy_min =
         ucs_cpu_memcpy_thresh(ucs_global_opts.arch.builtin_memcpy_min, 1 * UCS_KBYTE);
     ucs_global_opts.arch.builtin_memcpy_max =
-        ucs_cpu_memcpy_thresh(ucs_global_opts.arch.builtin_memcpy_max, 8 * UCS_MBYTE);
+        ucs_cpu_memcpy_thresh(ucs_global_opts.arch.builtin_memcpy_max,
+                              l3_cache ? l3_cache / 2 : UCS_MEMUNITS_INF);
```

Comment:
1. ( ) around /
2. document the reason for such logic
3. if no L3 cache, do we really want to set new memcpy limit to INF? maybe set some default limit, or disable it completely?

Reply:
1. arounded
2. added comment about best performance value
3. right, one of commit left in stash... fixed